x-pack/plugins/index_management/public/application/services/api.ts

```diff
@@ -194,15 +194,15 @@ export async function loadIndexSettings(indexName: string) {
 }
 
 export async function updateIndexSettings(indexName: string, body: object) {
-  const response = await httpService.httpClient.put(
-    `${API_BASE_PATH}/settings/${encodeURIComponent(indexName)}`,
-    {
-      body: JSON.stringify(body),
-    }
-  );
-  // Only track successful requests.
-  uiMetricService.trackMetric(METRIC_TYPE.COUNT, UIM_UPDATE_SETTINGS);
-  return response;
+  return sendRequest({
+    path: `${API_BASE_PATH}/settings/${encodeURIComponent(indexName)}`,
+    method: 'put',
+    body: JSON.stringify(body),
+  }).then((result) => {
+    // Only track successful requests.
+    uiMetricService.trackMetric(METRIC_TYPE.COUNT, UIM_UPDATE_SETTINGS);
```

Comment:
Looking at `sendRequest` it wraps the HTTP request behind a `try / catch`. This means that this Promise will always succeed. We should then wrap the `uiMetricService.trackMetric` call inside an

```js
if (result.error === null) { ... }
```

Reply:
Totally right!